src/leiningen/monolith.clj

```diff
@@ -274,28 +275,52 @@
     (fingerprint/clear project opts more)))
 
 
+(defn ^:higher-order with-dependency-set
+  "Run a task with a set of managed dependencies from a named dependency set.
+
+   Overrides the dependencies from the named dependency set into the project.
+   For the root project, this means the managed dependencies will be overwritten
+   with the dependencies from the named set. For subprojects, the
+   `:monolith/dependency-set` metadata key will be set to the named set. If the
+   project does not have the `:monolith/dependency-set` key defined, it will be
+   added.
```

Comment:
I think this last sentence is implied by the previous one, so you could shorten a bit here.